### ISSUE TITLE
IllegalStateException and ClassCastException Android vitals reporting Crashes and ANRs

### ISSUE THREAD

**Issue by emmanuelfrancis** [transistorsoft/react-native-background-geolocation#1952]
## Your Environment
* Plugin version:     "react-native-background-fetch": "^4.2.1",
    "react-native-background-geolocation": "^4.14.5",

* Platform: Android
* OS version: Android 9-14
* Device manufacturer / model: Samsung, Motorola, KYOCERA E6910, T-Mobile Sprout, TCL Bremen_TF
* React Native version (`react-native -v`): 0.71.11
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
App.js
unsubscribeOnGeofence = BackgroundGeolocation.onGeofence(onGeofenceHandler);
unsubscribeOnSchedule = BackgroundGeolocation.onSchedule(onGeoFenceScheduler);

    BackgroundGeolocation.ready({
      desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_MEDIUM,
      stopOnTerminate: false,
      startOnBoot: true,
      showsBackgroundLocationIndicator: true,
      locationAuthorizationRequest: 'Always',
      disableMotionActivityUpdates: false,
      activityType: BackgroundGeolocation.ACTIVITY_TYPE_OTHER,
      geofenceModeHighAccuracy: true,
      scheduleUseAlarmManager: true,
      logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
      enableHeadless: true,
      disableLocationAuthorizationAlert: true,
     notification: {
	        // some custom message
        },
        backgroundPermissionRationale: {
        	// some custom message
        },
        locationAuthorizationAlert: {
            // some custom message
        },
    })
      .then(state => {
        //console.log("ready state ", state)
      })
      .catch(error => {
        console.log('- BackgroundGeolocation error: ', error);
      });

index.js
const HeadlessTask = async event => {
  const {params} = event;
  switch (event.name) {
    case 'geofence':
      // onGeofenceHandler
      onGeofenceHandler(params);
      break;
    case 'schedule':
      onGeoFenceScheduler(params);
      break;
    default: {
      console.log('not matching');
    }
  }
};
BackgroundGeolocation.registerHeadlessTask(HeadlessTask);

// Based on server data, the app sets geofences and the schedules for android

BackgroundGeolocation.addGeofences(geofences, () => {
      console.log('addGeofences new called');
    });

const updateSchedule = []
 BackgroundGeolocation.setConfig({
    schedule: updateSchedule,
    scheduleUseAlarmManager: true,
  })
    .then(state1 => {
      if (updateSchedule && updateSchedule.length > 0) {
        BackgroundGeolocation.startSchedule().then(state2 => {
        });
      }
    })
    .catch(error => {
    });

export const onGeoFenceScheduler = async state => {
  const {enabled} = state;
  if (enabled) {
    const geofences = await BackgroundGeolocation.getGeofences();
    if (geofences.length > 0) {
      BackgroundGeolocation.setConfig({
        desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
      });
      BackgroundGeolocation.startGeofences(
        state2 => {
        },
        error => {
        },
      );
    }
  } else {
    BackgroundGeolocation.setConfig({
      desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_MEDIUM,
    });
    BackgroundGeolocation.stop();
  }
};
```

## Expected Behavior
Google Play store should not have Crashes

## Actual Behavior
Google Play Store reports Crashes

## Steps to Reproduce
In development or on my devices I do not see any crash in adb logs or any crashes. I have tested it on Samsung M14 and Nokia 5.3. My development devices are not listed in Google play store crash device lists.

## Context
Everything works fine as per the expected functionalities. The scheduler gets started as per configured schedule and stopped. Geo fences are also called properly on my devices.

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
Below are the stack trace from Google Play store - Crashes and ANRs
com.transistorsoft.locationmanager.logger.TSLog.a
Type java.lang.ClassCastException
Exception java.lang.ClassCastException: org.slf4j.helpers.SubstituteLogger cannot be cast to ch.qos.logback.classic.Logger
  at com.transistorsoft.locationmanager.logger.TSLog.a
  at com.transistorsoft.locationmanager.logger.TSLog.a
  at com.transistorsoft.locationmanager.logger.TSLog.$r8$lambda$3YB4RZFGV-hL2r0EClp1u86FGKM
  at com.transistorsoft.locationmanager.logger.TSLog$$ExternalSyntheticLambda1.run
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1145)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:644)
  at java.lang.Thread.run (Thread.java:1012)

com.transistorsoft.locationmanager.logger.TSSQLiteAppender$c.run
Type: java.lang.IllegalStateException
Exception java.lang.IllegalStateException:
  at android.database.sqlite.SQLiteClosable.acquireReference (SQLiteClosable.java:58)
  at android.database.sqlite.SQLiteDatabase.executeSql (SQLiteDatabase.java:2071)
  at android.database.sqlite.SQLiteDatabase.execSQL (SQLiteDatabase.java:2010)
  at com.transistorsoft.locationmanager.logger.TSSQLiteAppender$c.run
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1145)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:644)
  at java.lang.Thread.run (Thread.java:1012)

```

</details>

**Comment by christocracy** [comment]
Duplicate #1699.  Please *search* issues before posting.

**Comment by emmanuelfrancis** [comment]
Hi @christocracy As per the mentioned ticket  I have added 
`slf4jVersion = "1.7.36"
 logbackVersion = "2.0.1"`

But I am still getting below error:
`Exception java.lang.IllegalStateException: attempt to re-open an already-closed object: SQLiteDatabase: /data/user/0/com.xxx.xxx/databases/transistor_logback.db
  at android.database.sqlite.SQLiteClosable.acquireReference (SQLiteClosable.java:58)
  at android.database.sqlite.SQLiteDatabase.executeSql (SQLiteDatabase.java:2071)
  at android.database.sqlite.SQLiteDatabase.execSQL (SQLiteDatabase.java:2010)
  at com.transistorsoft.locationmanager.logger.TSSQLiteAppender$c.run
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1145)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:644)
  at java.lang.Thread.run (Thread.java:1012)`

Is this something different? I tried searching the issues but didn't find anything related to. Is there any other fix for this?
